File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int sumBlock = brown + yellow;
        
        for(int x = 3; x < sumBlock - 2; x++) {
            int y = sumBlock / x;
            if (brown == 2 * ((y - 2) + x)) {
                if (((x - 2) * (y - 2)) == yellow) {
                    answer[0] = x;
                    answer[1] = y;
                }
            }
        
        }
        
        return answer;
    }
}

File: Python3/프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    sum_block = brown + yellow
    x = 0
    
    for i in range(3, sum_block):
        x = sum_block // i
        if sum_block % i == 0 and 2 * (x + i - 2) == brown:
            return [x, i]
    